### PR TITLE
Fix possible null dereference in hlo_module_util

### PR DESCRIPTION
**Pull request by apach301**
At hlo_module_util.cc:
Possible null dereference of `execution_options`. Should be checked as in the line https://github.com/apach301/tensorflow/blob/356bc3788b40b2240244262758f7ca13dd73de64/tensorflow/compiler/xla/service/hlo_module_util.cc#L125

This PR is part of https://github.com/tensorflow/tensorflow/pull/57892

Bug was found by [Svace static analyzer](https://www.ispras.ru/en/technologies/svace/) (more [info](https://svace.pages.ispras.ru/svace-website/en/)).

### CODE REVIEW

**Comment by cheshire** [comment]
Is it possible to separate out the two changes and generate a test for the second one?

File: tensorflow/compiler/xla/service/hlo_module_util.cc [hunk 1]
```diff
@@ -129,8 +129,11 @@ StatusOr<std::unique_ptr<HloModuleConfig>> CreateModuleConfig(
         DeviceAssignment::Deserialize(execution_options->device_assignment()));
     config->set_static_device_assignment(*device_assignment);
   }
-  config->set_alias_passthrough_params(
```
Comment (cheshire) [inline tensorflow/compiler/xla/service/hlo_module_util.cc:132]: On line 97 it assumes that execution_options is non-null already, so I think we should remove the guard at line 125 instead (and probably your analyzer should highlight this guard as dead code instead, as it is guaranteed to be non-null as we reach it)

**Review by cheshire — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/hlo_module_util.cc (hunk 1 above)
Comment (apach301) [reply tensorflow/compiler/xla/service/hlo_module_util.cc:132]: Line 97 or 87? If you mean line 87, then it only ensures that execution_options is non-null up to the line 116. After that there is else-statement on lines 116-119 where control-flow is not interrupted. So execution_options still may be null on 125 and 132, unless I missed something

Comment (cheshire) [reply tensorflow/compiler/xla/service/hlo_module_util.cc:132]: Ah I see, thanks! Maybe then move both set_static_device_assignment and set_alias_passthrough_param under the guard starting at line 87 then?

Comment (apach301) [reply tensorflow/compiler/xla/service/hlo_module_util.cc:132]: Yeah, it would be more neat solution

**Review by cheshire — APPROVED** [review]

**Pull request merged**